tiledb/sm/query/writer.h

```diff
@@ -260,8 +282,42 @@ class Writer {
    * @return Status
    */
   Status set_buffer(
+      const std::string& name, void* const buffer, uint64_t* const buffer_size);
+
+  /**
+   * Sets the data buffer for a agnostic (fixed/var) sized attribute/dimension.
+   *
+   * @param name The attribute/dimension to set the buffer for.
+   * @param buffer The buffer that has the input data to be written.
+   * @param buffer_size The size of `buffer` in bytes.
+   * @return Status
+   */
+  Status set_buffer_data(
       const std::string& name, void* buffer, uint64_t* buffer_size);
 
+  /**
+   * Sets the offset buffer for a agnostic (fixed/var) sized
+   * attribute/dimension.
+   *
+   * @param name The attribute/dimension to set the buffer for.
+   * @param buffer The buffer that has the input data to be written.
+   * @param buffer_size The size of `buffer` in bytes.
+   * @return Status
+   */
+  Status set_buffer_offsets(
+      const std::string& name, uint64_t* buffer, uint64_t* buffer_size);
+
+  /**
+   * Sets the validity buffer for a agnostic sized nullable attribute/dimension.
+   *
+   * @param name The attribute/dimension to set the buffer for.
+   * @param buffer The buffer that has the input data to be written.
+   * @param buffer_size The size of `buffer` in bytes.
+   * @return Status
+   */
+  Status set_buffer_validity(
```

Comment:
Rename to `set_validity_buffer`.

Reply:
I changed these in `tiledb/sm/query/query_buffer.h` as well for the sake of consistency. Let me know if you want that change reverted.